web.browser/src/redux/reducer.js

```diff
@@ -0,0 +1,36 @@
+import { data } from '../mock-data';
+import {
+    SORT_BY_DATE,
+    SORT_BY_UPVOTE,
+    UPDATE_VOTE
+} from './actions';
+
+
+export function PostsReducer(state = data.posts, action) {
+  switch (action.type) {
+    case UPDATE_VOTE:
+        return state.map((post) => {
+            if (action.id === post.id) post.votes++;
+            return post;
+        });
```

Comment:
This is actually in impure function, since you're incrementing `post.votes`. To fix:

```js
return state.map((post) => (
  action.id === post.id ? post.votes + 1 : post.votes
));
```